### PR TITLE
Use has-signed-canonical-cla GitHub Action

### PR DESCRIPTION
**Pull request by MarcusTomlinson**
- [x] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] Have you successfully run `./runtests.sh static`?
- [x] Have you successfully run `./runtests.sh tests/unit`?

-----

### CODE REVIEW

**Comment by cjp256** [comment]
Excellent! I had recently updated https://github.com/kyrofa/cla-check to start consolidating this usage, a github action is much nicer! :D

IIUC, I have a concern with https://github.com/canonical/has-signed-canonical-cla/commit/2a38319a60d7ba858090b70a8e56b45a50736921. @sergiusens told me that trivial PRs can be merged without CLA approval.  If this allows all previous contributors, someone can get a trivial PR committed and appear signed in subsequent non-trivial PRs, which I do not believe is desirable.

File: .github/workflows/cla-check.yaml [hunk 1]
```diff
@@ -3,19 +3,7 @@ on: [pull_request]
 
 jobs:
   cla-check:
-    runs-on: ubuntu-18.04
+    runs-on: ubuntu-latest
```
Comment (cjp256) [inline .github/workflows/cla-check.yaml:6]:
```suggestion
        runs-on: ubuntu-20.04        
```

**Comment by MarcusTomlinson** [comment]
> Excellent! I had recently updated https://github.com/kyrofa/cla-check to start consolidating this usage, a github action is much nicer! :D
> 
> IIUC, I have a concern with [canonical/has-signed-canonical-cla@2a38319](https://github.com/canonical/has-signed-canonical-cla/commit/2a38319a60d7ba858090b70a8e56b45a50736921). @sergiusens told me that trivial PRs can be merged without CLA approval. If the action allows all previous contributors, someone can get a trivial PR committed and appear signed in subsequent non-trivial PRs, which I do not believe is desirable.

You make a very valid point. Default behaviour should not be to accept existing contributors, I agree.

I've [updated the action](https://github.com/marketplace/actions/has-signed-canonical-cla?version=1.1.1) to NOT accept existing contributors by default but instead take an optional input parameter called `accept-existing-contributors` to allow users to enable this feature should they wish to.

**Comment by cjp256** [comment]
> > Excellent! I had recently updated https://github.com/kyrofa/cla-check to start consolidating this usage, a github action is much nicer! :D
> > IIUC, I have a concern with [canonical/has-signed-canonical-cla@2a38319](https://github.com/canonical/has-signed-canonical-cla/commit/2a38319a60d7ba858090b70a8e56b45a50736921). @sergiusens told me that trivial PRs can be merged without CLA approval. If the action allows all previous contributors, someone can get a trivial PR committed and appear signed in subsequent non-trivial PRs, which I do not believe is desirable.
> 
> You make a very valid point. Default behaviour should not be to accept existing contributors, I agree.
> 
> I've [updated the action](https://github.com/marketplace/actions/has-signed-canonical-cla?version=1.1.1) to NOT accept existing contributors by default but instead take an optional input parameter: `accept-existing-contributors` to allow users to enable this feature should they wish to.

Great, thanks!!

**Review by cjp256 — APPROVED** [review]
LGTM

**Comment by sergiusens** [comment]
Would this discriminate the commits from what was already merged in main (results from clicking "Update branch")? Some of those commits would not have signed the CLA and this would then all of the sudden fail. I did not see that logic in the action, but it might be expressed somewhere else and I am just missing it.

**Comment by MarcusTomlinson** [comment]
> Would this discriminate the commits from what was already merged in main (results from clicking "Update branch")? Some of those commits would not have signed the CLA and this would then all of the sudden fail. I did not see that logic in the action, but it might be expressed somewhere else and I am just missing it.

The commits are pulled from the GitHub `pull_request` payload like so: https://github.com/canonical/has-signed-canonical-cla/blob/main/index.js#L39

These commits are formatted as they appear on the "Commits" tab of the PR. For example: https://github.com/canonical/ubuntu-desktop-installer/pull/44/commits

You'll see in the above example that a merge appears as a single commit authored by myself and not as all contained commits (as you would from calling something like `git log/shortlog`).

I think this addresses your concern?

**Comment by sergiusens** [comment]
Marcus Tomlinson writes:

>> Would this discriminate the commits from what was already merged in main (results from clicking "Update branch")? Some of those commits would not have signed the CLA and this would then all of the sudden fail. I did not see that logic in the action, but it might be expressed somewhere else and I am just missing it.
>
> The commits are pulled from the GitHub `pull_request` payload like so: https://github.com/canonical/has-signed-canonical-cla/blob/main/index.js#L39
>
> These commits are formatted as they appear on the "Commits" tab of the PR. For example: https://github.com/canonical/ubuntu-desktop-installer/pull/44/commits
>
> You'll see in the above example that a merge appears as a single commit authored by myself and not as all contained commits (as you would from calling something like `git log/shortlog`).
>
> I think this addresses your concern?

It does, thank you.

**Comment by MarcusTomlinson** [comment]
Is this ready to merge then?

**Pull request merged**